### PR TITLE
Make API use AWS Step Function activities

### PR DESCRIPTION
**Pull request by marcellp**
# Background
#### Link to issue 
https://biomage.atlassian.net/browse/BIOMAGE-709

#### Link to staging deployment URL 
TBD

#### Links to any Pull Requests related to this
https://github.com/biomage-ltd/pipeline/pull/28

#### Anything else the reviewers should know about the changes here
N/A

# Changes
### Code changes
API is slightly refactored to support activities. Only one job (old pipeline server) is launched, the "clients" are substituted with activities in AWS Step Functions.

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [ ] Unit tests written
- [x] Tested locally with Inframock
- [ ] Deployed to staging

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [ ] Photo of a cute animal attached to this PR

### CODE REVIEW

File: src/api/general-services/pipeline-manage/constructors/create-new-step.js [hunk 1]
```diff
@@ -12,117 +11,25 @@ const createNewStep = (context, step, args) => {
...
-
-  const stepHash = crypto.randomBytes(5).toString('hex');
-
-  if (config.clusterEnv === 'development') {
-    return {
-      ...step,
-      Type: 'Task',
-      Resource: 'arn:aws:states:::lambda:invoke',
-      // TODO: fix this, add some of old output
-      // see https://docs.aws.amazon.com/step-functions/latest/dg/input-output-resultpath.html
```
Comment (ivababukova) [inline src/api/general-services/pipeline-manage/constructors/create-new-step.js:30]: are we no longer going to use lambdas to run this locally?

File: src/api/general-services/pipeline-manage/constructors/create-new-step.js [hunk 2]
```diff
@@ -12,117 +11,25 @@ const createNewStep = (context, step, args) => {
...
-    };
-  }
-
+  };
 
   return {
     ...step,
     Type: 'Task',
-    Comment: 'Attempts to create a Kubernetes Job for the pipeline server. Will swallow a 409 (already exists) error.',
-    Resource: 'arn:aws:states:::eks:runJob.sync',
```
Comment (ivababukova) [inline src/api/general-services/pipeline-manage/constructors/create-new-step.js:57]: if this is no longer going to be an eks job, you should remember to delete the pipeline can spawn role for k8s in pipeline repo.

File: src/api/general-services/pipeline-manage/constructors/create-new-step.js (hunk 1 above)
Comment (marcellp) [reply src/api/general-services/pipeline-manage/constructors/create-new-step.js:30]: No, just for the runner itself. Rest is going to be activities.

File: src/api/general-services/pipeline-manage/constructors/create-new-step.js (hunk 2 above)
Comment (marcellp) [reply src/api/general-services/pipeline-manage/constructors/create-new-step.js:57]: We still have a kubernetes job step for spawning the actual server, so we can keep it.

File: src/api/general-services/pipeline-manage/constructors/create-new-step.js (hunk 1 above)
Comment (ivababukova) [reply src/api/general-services/pipeline-manage/constructors/create-new-step.js:30]: that sounds much better -- now we will have the running locally represent better what is actually happening

File: src/api/general-services/pipeline-manage/constructors/create-new-step.js (hunk 2 above)
Comment (ivababukova) [reply src/api/general-services/pipeline-manage/constructors/create-new-step.js:57]: good point, ok

File: src/api/general-services/pipeline-manage/constructors/create-new-step.js [hunk 3]
```diff
@@ -12,117 +11,24 @@ const createNewStep = (context, step, args) => {
...
-    // TODO: fix this, add some of old output
-    // see https://docs.aws.amazon.com/step-functions/latest/dg/input-output-resultpath.html
+    Resource: activityArn,
     ResultPath: null,
+    TimeoutSeconds: 300,
```
Comment (xverges) [inline src/api/general-services/pipeline-manage/constructors/create-new-step.js:26]: Good idea!

**Review by xverges — APPROVED** [review]

**Pull request merged**